Makefile.e2e.mk

```diff
@@ -170,32 +326,145 @@ apply/example/minikube/mtls: ## Minikube: enable mTLS
+	$(call kubectl_exec,kuma-demo,demo-app,kuma-sidecar) wget -qO- http://localhost:9901/stats/prometheus | grep upstream_rq_total
 
 verify/example/minikube/inbound:
-	$(call verify_example_inbound,kubectl -n kuma-demo exec $$(kubectl -n kuma-demo get pods -l app=demo-app -o=jsonpath='{.items[0].metadata.name}') -c kuma-sidecar -- )
+	$(call verify_example_inbound,$(call kubectl_exec,kuma-demo,demo-app,kuma-sidecar))
 
 verify/example/minikube/outbound:
-	$(call verify_example_outbound,kubectl -n kuma-demo exec $$(kubectl -n kuma-demo get pods -l app=demo-app -o=jsonpath='{.items[0].metadata.name}') -c kuma-sidecar -- )
+	$(call verify_example_outbound,$(call kubectl_exec,kuma-demo,demo-app,kuma-sidecar))
 
 verify/example/minikube: verify/example/minikube/inbound verify/example/minikube/outbound ## Minikube: Verify Envoy stats (after sample requests)
 
 verify/example/minikube/mtls: verify/example/minikube/mtls/outbound ## Minikube: Verify Envoy mTLS stats (after sample requests)
 
 verify/example/minikube/mtls/outbound:
 	@echo "Checking number of Outbound mTLS requests via Envoy ..."
-	test $$( kubectl -n kuma-demo exec $$(kubectl -n kuma-demo get pods -l app=demo-client -o=jsonpath='{.items[0].metadata.name}') -c kuma-sidecar -- wget -qO- http://localhost:9901/stats/prometheus | grep 'envoy_cluster_kuma_demo_svc_8000_ssl_handshake{envoy_cluster_name="demo-app"}' | awk '{print $$2}' | tr -d [:space:] ) -ge 5
+	test $$( $(call kubectl_exec,kuma-demo,demo-client,kuma-sidecar) wget -qO- http://localhost:9901/stats/prometheus | grep 'envoy_cluster_kuma_demo_svc_8000_ssl_handshake{envoy_cluster_name="demo-app"}' | awk '{print $$2}' | tr -d [:space:] ) -ge 5
 	@echo "Check passed!"
 
 kumactl/example/minikube:
 	cat tools/e2e/examples/minikube/kumactl_workflow.sh | docker run -i --rm --user $$(id -u):$$(id -g) --network host -v $$HOME/.kube:/tmp/.kube -v $$HOME/.minikube:$$HOME/.minikube -e HOME=/tmp -w /tmp $(KUMACTL_DOCKER_IMAGE)
+
+undeploy/example/minikube: ## Minikube: Undeploy example setup
+	kubectl delete -f tools/e2e/examples/minikube/kuma-demo/
+
+deploy/traffic-routing/minikube: ## Minikube: Deploy example setup for TrafficRoute
+	@echo
+	@echo "Deploying example setup for TrafficRoute ..."
+	@echo
+	kubectl apply -f tools/e2e/examples/minikube/kuma-routing/
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-web
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-web
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-backend-v1
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-backend,version=v1
+	kubectl wait --timeout=60s --for=condition=Available -n kuma-example deployment/kuma-example-backend-v2
+	kubectl wait --timeout=60s --for=condition=Ready -n kuma-example pods -l app=kuma-example-backend,version=v2
```

Comment:
(don't change this it's fine) I'm a fan of
```
while [[ "$(kubectl get pod --all-namespaces | grep -v Running | grep -v Completed | wc -l)" != 1 ]]; do
  kubectl get pod --all-namespaces -o wide
  echo "waiting for K8s to be ready"
  sleep 10;
done
```

It's less specific but sometimes that's what's needed ;)

Reply:
This `while` loop doesn't account for a situation where `Pods` are not created at all